### ISSUE TITLE
Tests no longer passing on Modulise branch

### ISSUE THREAD

**Issue by shakna-israel** [shakna-israel/docbuilder#34]
Though it appears to compile correctly

**Comment by shakna-israel** [comment]
Traceback:

```
F.F.EEEEEEEEF
======================================================================
ERROR: Test that bolds appear correctly when compiled to Markdown.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 161, in test_markdown_bolds
    buildup()
  File "/home/pi/docbuilder/tests/test.py", line 30, in buildup
    testfile=open('testdocs/testme.md')
IOError: [Errno 2] No such file or directory: 'testdocs/testme.md'

======================================================================
ERROR: Test that bullet points appear correctly when compiled to Markdown.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 133, in test_markdown_bullet_points
    buildup()
  File "/home/pi/docbuilder/tests/test.py", line 30, in buildup
    testfile=open('testdocs/testme.md')
IOError: [Errno 2] No such file or directory: 'testdocs/testme.md'

======================================================================
ERROR: Test that codeblocks are correctly compiled to Markdown.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 189, in test_markdown_codeblocks
    buildup()
  File "/home/pi/docbuilder/tests/test.py", line 30, in buildup
    testfile=open('testdocs/testme.md')
IOError: [Errno 2] No such file or directory: 'testdocs/testme.md'

======================================================================
ERROR: Test that normal comments appear correctly when compiled to Markdown.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 119, in test_markdown_comments
    buildup()
  File "/home/pi/docbuilder/tests/test.py", line 30, in buildup
    testfile=open('testdocs/testme.md')
IOError: [Errno 2] No such file or directory: 'testdocs/testme.md'

======================================================================
ERROR: Test to see if code blocks indent correctly when compiled to Markdown
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 216, in test_markdown_indented_code_block
    buildup()
  File "/home/pi/docbuilder/tests/test.py", line 30, in buildup
    testfile=open('testdocs/testme.md')
IOError: [Errno 2] No such file or directory: 'testdocs/testme.md'

======================================================================
ERROR: Test that italics appear correctly when compiled to Markdown.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 147, in test_markdown_italics
    buildup()
  File "/home/pi/docbuilder/tests/test.py", line 30, in buildup
    testfile=open('testdocs/testme.md')
IOError: [Errno 2] No such file or directory: 'testdocs/testme.md'

======================================================================
ERROR: Test that words that are both bold and italic appear correctly when compiled to Markdown.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 175, in test_markdown_italics_and_bolds
    buildup()
  File "/home/pi/docbuilder/tests/test.py", line 30, in buildup
    testfile=open('testdocs/testme.md')
IOError: [Errno 2] No such file or directory: 'testdocs/testme.md'

======================================================================
ERROR: Test that titles appear correctly when compiled to Markdown.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 104, in test_markdown_titles
    buildup()
  File "/home/pi/docbuilder/tests/test.py", line 30, in buildup
    testfile=open('testdocs/testme.md')
IOError: [Errno 2] No such file or directory: 'testdocs/testme.md'

======================================================================
FAIL: Test that Docbuilder can make custom documentation directories
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 99, in test_custom_directories
    assert False
AssertionError:
-------------------- >> begin captured stdout << ---------------------
Docbuilder failed to create a custom directory.

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: Test to test that Docbuilder behaves correctly when given both a file to document, and a filename for where to place the documentation.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 86, in test_file_given
    assert False
AssertionError:
-------------------- >> begin captured stdout << ---------------------
Docbuilder failed to build correctly when given both a from filename and a to filename.

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: Test that Docbuilder does comply with PEP8.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/docbuilder/tests/test.py", line 45, in test_pep8_complicance
    "Found code style errors (and warnings).")
AssertionError: Found code style errors (and warnings).
-------------------- >> begin captured stdout << ---------------------
docbuilder.py:24:1: E302 expected 2 blank lines, found 1
docbuilder.py:36:1: E302 expected 2 blank lines, found 1
docbuilder.py:40:1: E302 expected 2 blank lines, found 1
docbuilder.py:44:1: E302 expected 2 blank lines, found 1
docbuilder.py:51:1: E302 expected 2 blank lines, found 1
docbuilder.py:58:1: E302 expected 2 blank lines, found 1
docbuilder.py:65:1: E302 expected 2 blank lines, found 1
docbuilder.py:69:1: E302 expected 2 blank lines, found 1
docbuilder.py:85:1: E302 expected 2 blank lines, found 1
docbuilder.py:102:1: E302 expected 2 blank lines, found 1
docbuilder.py:108:1: W293 blank line contains whitespace
docbuilder.py:109:15: E225 missing whitespace around operator
docbuilder.py:109:25: E231 missing whitespace after ':'
docbuilder.py:109:25: E701 multiple statements on one line (colon)

--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 13 tests in 3.162s

FAILED (errors=8, failures=3)
```

**Comment by shakna-israel** [comment]
Error appears to be with `tests/test.py` rather than with Docbuilder.

The `buildup()` function isn't running correctly.

**Comment by shakna-israel** [comment]
Further debugged: Docbuilder isn't building into a custom directory, so the testme.py file generated in testing ends up in the wrong directory.

**Comment by shakna-israel** [comment]
Further debugged: Docbuilder _does_ build the directory, but _does not_ build into it.

eg. `./docbuilder.py tester.py tester.md testDir`

Created:

`tester.md` in the same directory as `docbuilder.py` _and_ created `testDir` directory.

**Comment by shakna-israel** [comment]
As of ac2b76d4232275d5ff9d228be85dbb081d9706cc, Docbuilder builds to the correct directory.

However, it doesn't strip the `#` from the line. (See the [build](https://travis-ci.org/shakna-israel/docbuilder/builds/58584973)).

**Comment by shakna-israel** [comment]
As of 629827d4f186f19f2913c43d043bf4f12ff90149, Docbuilder passes all tests except PEP8.

See the [build](https://travis-ci.org/shakna-israel/docbuilder/builds/58586867).

**Comment by shakna-israel** [comment]
Log from `pep8 --show-source --show-pep8 --ignore=E266,E501 docbuilder.py`:

```
docbuilder.py:24:1: E302 expected 2 blank lines, found 1
def stringManage(lineInFile):
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:24:30: W291 trailing whitespace
def stringManage(lineInFile):
                             ^
    Trailing whitespace is superfluous.

    The warning returned varies on whether the line itself is blank, for easier
    filtering for those who want to indent their blank lines.

    Okay: spam(1)\n#
    W291: spam(1) \n#
    W293: class Foo(object):\n    \n    bang = 12
docbuilder.py:40:1: E302 expected 2 blank lines, found 1
def checkExportFile(fileExists):
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:44:1: E302 expected 2 blank lines, found 1
def checkExportDir(dir):
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:48:1: E302 expected 2 blank lines, found 1
def unicodeCompareChar(uniCode):
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:55:1: E302 expected 2 blank lines, found 1
def markdownWrite(stringLine, fileToWrite):
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:62:1: E302 expected 2 blank lines, found 1
def codeblockWrite(stringLine, fileToWrite):
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:69:1: E302 expected 2 blank lines, found 1
def initFileOut(outFile):
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:73:1: E302 expected 2 blank lines, found 1
def readFile(inputFile, outputFile):
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:89:1: E302 expected 2 blank lines, found 1
def getFlags():
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:106:1: E302 expected 2 blank lines, found 1
def main():
^
    Separate top-level function and class definitions with two blank lines.

    Method definitions inside a class are separated by a single blank line.

    Extra blank lines may be used (sparingly) to separate groups of related
    functions.  Blank lines may be omitted between a bunch of related
    one-liners (e.g. a set of dummy implementations).

    Use blank lines in functions, sparingly, to indicate logical sections.

    Okay: def a():\n    pass\n\n\ndef b():\n    pass
    Okay: def a():\n    pass\n\n\n# Foo\n# Bar\n\ndef b():\n    pass

    E301: class Foo:\n    b = 0\n    def bar():\n        pass
    E302: def a():\n    pass\n\ndef b(n):\n    pass
    E303: def a():\n    pass\n\n\n\ndef b(n):\n    pass
    E303: def a():\n\n\n\n    pass
    E304: @decorator\n\ndef a():\n    pass
docbuilder.py:112:1: W293 blank line contains whitespace

^
    Trailing whitespace is superfluous.

    The warning returned varies on whether the line itself is blank, for easier
    filtering for those who want to indent their blank lines.

    Okay: spam(1)\n#
    W291: spam(1) \n#
    W293: class Foo(object):\n    \n    bang = 12
docbuilder.py:113:15: E225 missing whitespace around operator
if __name__ =='__main__':main()
              ^
    Surround operators with a single space on either side.

    - Always surround these binary operators with a single space on
      either side: assignment (=), augmented assignment (+=, -= etc.),
      comparisons (==, <, >, !=, <=, >=, in, not in, is, is not),
      Booleans (and, or, not).

    - If operators with different priorities are used, consider adding
      whitespace around the operators with the lowest priorities.

    Okay: i = i + 1
    Okay: submitted += 1
    Okay: x = x * 2 - 1
    Okay: hypot2 = x * x + y * y
    Okay: c = (a + b) * (a - b)
    Okay: foo(bar, key='word', *args, **kwargs)
    Okay: alpha[:-i]

    E225: i=i+1
    E225: submitted +=1
    E225: x = x /2 - 1
    E225: z = x **y
    E226: c = (a+b) * (a-b)
    E226: hypot2 = x*x + y*y
    E227: c = a|b
    E228: msg = fmt%(errno, errmsg)
docbuilder.py:113:25: E231 missing whitespace after ':'
if __name__ =='__main__':main()
                        ^
    Each comma, semicolon or colon should be followed by whitespace.

    Okay: [a, b]
    Okay: (3,)
    Okay: a[1:4]
    Okay: a[:4]
    Okay: a[1:]
    Okay: a[1:4:2]
    E231: ['a','b']
    E231: foo(bar,baz)
    E231: [{'a':'b'}]
docbuilder.py:113:25: E701 multiple statements on one line (colon)
if __name__ =='__main__':main()
                        ^
    Compound statements (on the same line) are generally discouraged.

    While sometimes it's okay to put an if/for/while with a small body
    on the same line, never do this for multi-clause statements.
    Also avoid folding such long lines!

    Always use a def statement instead of an assignment statement that
    binds a lambda expression directly to a name.

    Okay: if foo == 'blah':\n    do_blah_thing()
    Okay: do_one()
    Okay: do_two()
    Okay: do_three()

    E701: if foo == 'blah': do_blah_thing()
    E701: for x in lst: total += x
    E701: while t < 10: t = delay()
    E701: if foo == 'blah': do_blah_thing()
    E701: else: do_non_blah_thing()
    E701: try: something()
    E701: finally: cleanup()
    E701: if foo == 'blah': one(); two(); three()
    E702: do_one(); do_two(); do_three()
    E703: do_four();  # useless semicolon
    E704: def f(x): return 2*x
    E731: f = lambda x: 2*x

```

**Comment by shakna-israel** [comment]
Solved as of ad401204dd8bffa779e380cdc54d95d24b0cca09
